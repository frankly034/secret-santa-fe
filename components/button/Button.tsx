import styles from "./button.module.css";

import React, { ComponentProps, FunctionComponent, ReactNode } from "react";
import { Button as AntButton } from "antd";

declare const ButtonTypes: readonly ["primary", "secondary"];

type ButtonType = (typeof ButtonTypes)[number];

interface ButtonProps extends Omit<ComponentProps<typeof AntButton>, "type"> {
  children: ReactNode;
  type?: ButtonType;
}

const Button: FunctionComponent<ButtonProps> = ({
  block = true,
  children,
  type = "primary",
  ...props
}) => {
  return (
    <AntButton block={block} className={styles[type]} shape="round" {...props}>
      {children}
    </AntButton>
  );
};

export default Button;
