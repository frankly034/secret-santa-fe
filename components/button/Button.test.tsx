import { fireEvent, render } from "@testing-library/react";

import Button from ".";

describe("Component - Button", () => {
  let onClick: () => void;

  beforeEach(() => {
    onClick = jest.fn();
  });

  it("renders button component", () => {
    const { container } = render(<Button>{"Click Me!"}</Button>);
    expect(container).toMatchSnapshot();
  });

  it("should render the default button with primary button class", () => {
    const { getByRole } = render(<Button>{"Click Me!"}</Button>);
    expect(getByRole("button").classList).toContain("primary");
  });

  it("should render the secondary button with secondary class", () => {
    const { getByRole } = render(
      <Button type="secondary">{"Click Me!"}</Button>
    );
    expect(getByRole("button").classList).toContain("secondary");
  });

  it("should fire the onClick function when clicked for active button", () => {
    const { getByRole } = render(
      <Button onClick={onClick} type="primary">
        {"Click Me!"}
      </Button>
    );

    fireEvent.click(getByRole("button"));
    expect(onClick).toHaveBeenCalledTimes(1);
  });

  it("should not fire the onClick function for disabled buttons", () => {
    const { getByRole } = render(
      <Button onClick={onClick} type="primary" disabled>
        {"Click Me!"}
      </Button>
    );

    fireEvent.click(getByRole("button"));
    expect(onClick).toHaveBeenCalledTimes(0);
  });
});
