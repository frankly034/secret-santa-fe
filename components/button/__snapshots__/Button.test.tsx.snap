// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - Button renders button component 1`] = `
<div>
  <button
    class="ant-btn css-dev-only-do-not-override-15rg2km ant-btn-round ant-btn-default ant-btn-block primary"
    type="button"
  >
    <span>
      Click Me!
    </span>
  </button>
</div>
`;
